sdk/servicebus/service-bus/src/core/streamingReceiver.ts

```diff
@@ -329,6 +344,16 @@ export class StreamingReceiver extends MessageReceiver {
           });
         }
       }
+
+      if (this.receiveMode === "peekLock" && numberOfEmptyIncomingSlots(this.link) <= 1) {
+        // Wait for the user to clear the deliveries before adding more credits
+        while (numberOfEmptyIncomingSlots(this.link) <= 1) {
+          await delay(1000);
+          if (numberOfEmptyIncomingSlots(this.link) > 1) {
+            this._receiverHelper.addCredit(1);
+          }
+        }
```

Comment:
Why do double calls to numberOfEmptyIncomingSlots with similar checks?
```suggestion
        while (numberOfEmptyIncomingSlots(this.link) <= 1) {
          await delay(1000);
        }
        this._receiverHelper.addCredit(1);
```

Reply:
Can be done!